kubectl-cloudflow/deploy/deploy.go

```diff
@@ -92,6 +86,331 @@ func GetCloudflowApplicationDescriptorFromDockerImage(dockerRegistryURL string,
 	return spec, *pulledImage
 }
 
+// CreateApplicationSpecFromBlueprintAndImages pulls all images necessary to create a Cloudflow Application descriptor
+// from a docker label. The function returns the created application spec and the pulled in images
+func CreateApplicationSpecFromBlueprintAndImages(imageRefs []ImageReference,
+	blueprintConfig *configuration.Config, replicas map[string]int) (domain.CloudflowApplicationSpec, []*docker.PulledImage) {
+
+	client, err := docker.GetVersionedClient()
+	if err != nil {
+		util.LogAndExit(err.Error())
+	}
+
+	// get all streamlet descriptors, image digests and pulled images in arrays
+	streamletDescriptors, imageDigests, apiVersion, pulledImages, deployImages, err := getStreamletDescriptorsAndImageInformation(imageRefs, client)
+	if err != nil {
+		util.LogAndExit(err.Error())
+	}
+
+	// load the blueprint
+	bp := makeBlueprint(blueprintConfig, imageDigests)
+
+	// Spec.Streamlets & Spec.Deployments
+	streamlets, deployments := getStreamletsAndDeployments(streamletDescriptors, deployImages, replicas, bp)
+
+	// Spec.Connections
+	conns := getConnections(bp)
+
+	// create the application spec
+	spec := makeApplicationSpec(bp, apiVersion, conns, streamlets, deployments)
+
+	return spec, pulledImages
+}
+
+func makeApplicationSpec(bp blueprint, apiVersion string, conns []domain.Connection, streamlets []domain.Streamlet,
+	deployments []domain.Deployment) domain.CloudflowApplicationSpec {
+
+	var spec domain.CloudflowApplicationSpec
+	spec.AgentPaths = map[string]string{
+		"prometheus": "/prometheus/jmx_prometheus_javaagent-0.11.0.jar",
+	}
+	spec.AppID = bp.name
+	spec.Version = apiVersion
+	spec.LibraryVersion = domain.LibraryVersion
+	spec.Connections = conns
+	spec.Streamlets = streamlets
+	spec.Deployments = deployments
+	return spec
+}
+
+func getConnections(bp blueprint) []domain.Connection {
+	var conns []domain.Connection
+	for inlet, outlets := range bp.connections {
+		for _, outlet := range outlets {
+			// need to switch inlet and outlet as we have them reversed in the blueprint map of connections
```

Comment:
Not sure what this comment means? something to still fix?
Or does it say what it does? (the "need to " throws me off)

Reply:
removed the comment